core/broadcast.go

```diff
@@ -116,7 +120,7 @@ func (b *echoBroadcast) BroadcastDKG(c context.Context, p *drand.DKGPacket) erro
 	defer b.Unlock()
 
 	addr := net.RemoteAddress(c)
-	dkgPacket, err := protoToDKGPacket(p.GetDkg())
+	dkgPacket, err := protoToDKGPacket(p.GetDkg(), b.scheme)
```

Comment:
Also, you only pass the scheme to the broadcast because of this line it seems, but in the end the protobuf conversion only needs the group, it doesn't need the full scheme.

Reply:
We could also imagine having an interface that takes care of serializing/deserializing these packets, and then the scheme could implement this interface and you would only pass this new interface to broadcast, which makes sense then. Right now the separation of concern is low.